pkg/db/db_test.go

```diff
@@ -129,6 +135,27 @@ func TestClient_NeedsUpdate(t *testing.T) {
 			},
 			want: false,
 		},
+		{
+			name: "DownloadedAt is zero, skip is false",
+			skip: false,
+			metadata: metadata.Metadata{
+				Version:      db.SchemaVersion,
+				DownloadedAt: time.Time{}, // zero time
+				NextUpdate:   timeNextUpdateDay1,
+			},
+			want: true,
+		},
+		{
+			name: "DownloadedAt is zero, skip is true",
```

Comment:
What if a user downloads the database manually under an air-gapped environment?
https://trivy.dev/latest/docs/advanced/self-hosting/

Reply:
Nice catch!
For DB got using `oras` DownloadAt` always will be zero.

I updated logic, added warning and updated PR description - https://github.com/aquasecurity/trivy/pull/8864/commits/68b1dae170e2f4971c8fa7cf8bd32e9867b2d34e